Functions/Administration/xASL_adm_DefineASLSequence.m

```diff
@@ -5,87 +5,86 @@
+if isfield(xQ, 'PulseSequenceType')
+    if isempty(regexpi(xQ.PulseSequenceType, '(epi|grase|spiral)', 'once'))
+        warning(['Unknown ASL sequence: ' xASL_num2str(xQ.PulseSequenceType)]);
         fprintf('%s\n', 'Trying to fix this');
-        xQ = rmfield(xQ, 'Sequence');
+        xQ = rmfield(xQ, 'PulseSequenceType');
 
     end
 end
 
 %% Check vendor field
 % Assume vendor=GE for 3D spiral sequences 
-% (though this is tricky for special cases, Siemens and Philips are both
-% working on a 3D spiral)
-if isfield(xQ,'Sequence') && (~isfield(xQ, 'Vendor') || isempty(xQ.Vendor))
-    if strcmpi(xQ.Sequence,'3d_spiral')
+% (though this is tricky for special cases, Siemens and Philips are both working on a 3D spiral)
+if isfield(xQ, 'PulseSequenceType') && (~isfield(xQ, 'Vendor') || isempty(xQ.Vendor))
+    if strcmpi(xQ.PulseSequenceType,'spiral')
         warning('xQ.Vendor missing but 3D spiral sequence detected, assuming vendor GE');
         xQ.Vendor = 'GE';
     end
 end
 
 if ~isfield(xQ, 'Vendor') || isempty(xQ.Vendor)
     warning('xQ.Vendor missing, skipping determining ASL sequence');
-elseif  isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms|GE|Philips|Siemens'))
+elseif  isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms|GE|Philips|Siemens', 'once'))
     warning('Unknown Vendor specified in xQ.Vendor');
-elseif ~isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms'))
+elseif ~isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms', 'once'))
     fprintf('%s\n', 'Digital Reference Object ASL-DRO detected');
 end
 
 
 %% Try to work out which ASL sequence we have
 % First assume that 2D is 2D EPI, irrespective of Vendor
-if ~isfield(xQ, 'Sequence') && isfield(xQ, 'readoutDim') 
-	if strcmpi(xQ.readoutDim,'2D')
-		xQ.Sequence = '2D_EPI';
+if ~isfield(xQ, 'PulseSequenceType') && isfield(xQ, 'MRAcquisitionType') 
+	if strcmpi(xQ.MRAcquisitionType, '2D')
+		xQ.PulseSequenceType = 'EPI';
         if bVerbose
-            if ~isempty(regexpi(xQ.Vendor,'Gold Standard Phantoms'))
+            if ~isempty(regexpi(xQ.Vendor, 'Gold Standard Phantoms', 'once'))
 			    fprintf('%s\n', 'Processing as if this is a 2D EPI sequence');
 			    fprintf('%s\n', 'Though the acquisition is not simulated, this will assume acquisition of multi-slice 2D acquisitions');
 			    fprintf('%s\n', 'and heavy geometric distortion and minimal smoothness');
 		    else
 			    fprintf('%s\n', '2D readout detected, assuming 2D EPI');
             end
         end
-	elseif isfield(xQ, 'Vendor') && strcmpi(xQ.readoutDim,'3D')
-		if  ~isempty(regexpi(xQ.Vendor, 'Philips')) || ~isempty(regexpi(xQ.Vendor, 'Siemens'))
-			xQ.Sequence = '3D_GRASE'; % assume that 3D Philips or Siemens is 3D GRASE
+	elseif isfield(xQ, 'Vendor') && strcmpi(xQ.MRAcquisitionType, '3D')
+		if  ~isempty(regexpi(xQ.Vendor, 'Philips', 'once')) || ~isempty(regexpi(xQ.Vendor, 'Siemens', 'once'))
+			xQ.PulseSequenceType = 'GRASE'; % assume that 3D Philips or Siemens is 3D GRASE
```

Comment:
Again, I would make all lowercase, is more consistent and perhaps also BIDS compliant.

Reply:
3D and GRASE are BIDS compliant.